app/operations/mec_check/initiate_application_mec_checks.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+require 'dry/monads'
+require 'dry/monads/do'
+require 'aca_entities/medicaid/mec_check'
+
+module MecCheck
+  # Take an application payload and add mec check results as evidence to applicants
+  class InitiateApplicationMecChecks
+    send(:include, Dry::Monads[:result, :do])
+
+    # @param [Hash] params
+    # @return [Dry::Result]
+    def call(params)
+      validated_params = yield validate_params(params)
+      application_payload = JSON.parse(validated_params[:payload])
+      job = yield find_or_create_job(validated_params[:message_id])
+      application_response = yield get_applicant_checks(application_payload, job)
+      publish_to_enroll(application_response)
+    end
+
+    protected
+
+    def validate_params(params)
+      return Failure({ error: "MEC Check feature not enabled." }) unless MedicaidGatewayRegistry[:mec_check].enabled?
+      return Failure({ error: "Payload cannot be empty" }) if params[:payload].blank?
+
+      Success(params)
+    end
+
+    def find_or_create_job(message_id)
+      result = Transmittable::FindOrCreateJob.new.call(message_id: message_id,
+                                                       key: :application_mec_check_request,
+                                                       started_at: DateTime.now,
+                                                       publish_on: DateTime.now)
+
+      if result.success?
+        @job = result.value!
+        Success(@job)
+      else
+        Failure('Unable to create job')
+      end
+    end
+
+    def get_applicant_checks(application_payload, job)
+      people = application_payload["applicants"]
+      people.each do |person|
+        person_mec_check = get_person_mec_check(person, job, application_payload["hbx_id"])
+        next unless person_mec_check.success?
+        evidence = person["local_mec_evidence"]
+        evidence["request_results"] = [person_mec_check.value!]
+        evidence["aasm_state"] = generate_aasm_state(person_mec_check.value!)
+      end
+      Success(application_payload)
+    rescue StandardError => e
+      Failure({ job_id: job.id, hbx_id: application_payload["hbx_id"], error: "Mec check failure => #{e}" })
```

Comment:
have a failure here but not saving to the job

Reply:
Probably a good place for job error, thanks.